File: parameters.py
"""General"""
# The simulation step
DELTA_T = 0.1


"""Environment"""
# The door will close DOOR_TIMEOUT seconds after the last 
# person who entered or left the environment 
DOOR_TIMEOUT = 6
# The minimum time the door will remain open
MIN_DOOR_TIME = 10
# The time required to open the door after it is signaled to open
DOOR_OPEN_DELAY = 1.5


"""Passenger"""
# The probability to select a seat instead of a standing spot
SEAT_SELECTION_PROB = 0.75
# The average speed of each passenger (m/s)
SPEED = 0.65
# The standard deviation of passengers' speeds
SPEED_STD = 0.12

# Affects how often standing passengers will notice that a seat has been freed 
# Average passenger patience in seconds
PATIENCE = 50 
# The standard deviation of passengers' patience
PATIENCE_STD = 5


"""Crowd Simulation (See Helbing and Molnár 1998 (Social Force Model))"""
# Max speed = MAX_SPEED_MULTIPLIER * SPEED
MAX_SPEED_MULTIPLIER = 1.3
# Angle of Sight (degrees)
TWOPHI = 200
# Out of View factor
C = 0.5
# Relaxation Time
TAU = 0.5

# == Ped-Ped Potential ==

V0 = 1.3
PEDPED_SIGMA = 0.25

# == Environment Obstacles Potential ==

U0_E = 10
R_E = 0.05

# == Standing Boundaries Potential ==

U0_SB = 2
R_SB = 0.05

# == Additional Factors (Not Included in the original model) ==

# Used to help passengers enter/exit without getting stuck 
# Acceleration Force Multiplier for passengers entering the environment
AFM_ENT = 1.3
# Acceleration Force Multiplier for passengers exiting the environment
AFM_EXIT = 1.6
# Acceleration Force Multiplier for passengers standing
AFM_STAND = 0.4


"""Transmission Simulation"""
# Social Distance Index (Pd) 
# Logarithmic Regression Factors 
# (ALPHA * ln(x) + BETA)/100 where x is distance in meters
ALPHA = -22.8
BETA = 51.16

# Transmission factor (quantum/s)
TF = 1.61 * 10**-4


"""Seat Utility"""
# The utility of seats with passengers on the adjacent seats is multiplied by MADJ
MADJ = 0.5
# The utility of window seats is multiplied by MW 
MW = 3
# Larger SIGMA_SEAT results in larger utility scores for seats further away   
SIGMA_SEAT = 1
# Max selection distance (m)
MAX_DIST = 5


"""Standing Spot Utility"""
# A larger SIGMA_SPOT results in larger utility scores for further away spots  
SIGMA_SPOT = 2

File: cpp/scenario/parameters.cpp
#include "parameters.h"

#include <string>
#include <map>

#include "../io/readwrite.h"


std::map<std::string, double> parameters = read_parameters();


extern const double DELTA_T = parameters["DELTA_T"];
extern const double DOOR_TIMEOUT = parameters["DOOR_TIMEOUT"];
extern const double MIN_DOOR_TIME = parameters["MIN_DOOR_TIME"];
extern const double DOOR_OPEN_DELAY = parameters["DOOR_OPEN_DELAY"];
extern const double SEAT_SELECTION_PROB = parameters["SEAT_SELECTION_PROB"];
extern const double SPEED = parameters["SPEED"];
extern const double SPEED_STD = parameters["SPEED_STD"];

extern const double PATIENCE = parameters["PATIENCE"];
extern const double PATIENCE_STD = parameters["PATIENCE_STD"];

extern const double MAX_SPEED_MULTIPLIER = parameters["MAX_SPEED_MULTIPLIER"];
extern const double TWOPHI = parameters["TWOPHI"];
extern const double C = parameters["C"];
extern const double TAU = parameters["TAU"];

extern const double V0 = parameters["V0"];
extern const double PEDPED_SIGMA = parameters["PEDPED_SIGMA"];

extern const double U0_E = parameters["U0_E"];
extern const double R_E = parameters["R_E"];

extern const double U0_SB = parameters["U0_SB"];
extern const double R_SB = parameters["R_SB"];

extern const double AFM_ENT = parameters["AFM_ENT"];
extern const double AFM_EXIT = parameters["AFM_EXIT"];
extern const double AFM_STAND = parameters["AFM_STAND"];

extern const double ALPHA = parameters["ALPHA"];
extern const double BETA = parameters["BETA"];

extern const double TF = parameters["TF"];

extern const double MADJ = parameters["MADJ"];
extern const double MW = parameters["MW"];
extern const double SIGMA_SEAT = parameters["SIGMA_SEAT"];
extern const double MAX_DIST = parameters["MAX_DIST"];

extern const double SIGMA_SPOT = parameters["SIGMA_SPOT"];
